### PR TITLE
Interactive list

### PR DESCRIPTION
**Pull request by MostafaShahat2021**
### In this project we :

- Add a function for adding a new task.
- Add a function for deleting a task.
- Add a function for editing task descriptions.
- Add function for Updating items object's value for completed key upon user actions.
- Add function for clear all button
- Use webpack to bundle JavaScript.
- Use NPM.
- Use webpack.
- Use webpack-cli.
- Use css-loader.
- Use style-loader.
- Modify webpack.config.js
- use HtmlWebpackPlugin.
- Setup local dev server
- Add Linters for HTML/CSS.
- Use best practices for HTML & CSS.
- Update README file.
- Add Mark up to the .html file.
- Add style to the website.

### CODE REVIEW

File: modules/interactive-list.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+const clearCompletedBtn = document.querySelector('.clear');
+
+clearCompletedBtn.addEventListener('click', () => {
+  tasks = tasks.filter((task) => !task.completed);
+  tasks.forEach((e, i) => {
+    e.index = i + 1;
+  });
+  localStorage.setItem('task-list', JSON.stringify(tasks));
+  addElemToPage();
+});
```
Comment (Shedrack-Sunday) [inline modules/interactive-list.js:30]: - [ ] Nice work you did here. however, the function does not clear the tasks completed. kindly fix this bug to complement the fantastic job you've done so far 👏 

## View :
image

**Review by Shedrack-Sunday — CHANGES_REQUESTED** [review]
Hi @MostafaShahat2021      👋 

# CHANGES REQUIRED ♻️

Great work on the project 👏
You've done well implementing the requirements,  However, there are a few you need to work on, to move to the next project. 💯

 



Highlights
✅All linters are checked
✅ Nice pull request description 
  
Required Changes ♻️

- [x] Nice efforts so far. The little changes let have to do with making the completed deletion effect on the task interactive based on user actions 

_check the comment section for a detailed review_ 

# Optional suggestions 

Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application.

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.

As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: modules/interactive-list.js (hunk 1 above)
Comment (MostafaShahat2021) [reply modules/interactive-list.js:30]: Hi @Shedrack-Sunday, Thanks for your efforts in reviewing my code 😄 .

**Review by ShadyShawkat — APPROVED** [review]
## Hi @MostafaShahat2021, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## To Highlight
- All linters are passing ✔️ 
- Descriptive commit messages ✔️ 
- Professional looking pull request ✔️  

Image

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by MostafaShahat2021** [comment]
Hi @ShadyShawkat, Thanks a lot 💯 for your efforts in reviewing my code, I really appreciate it 😃.

**Pull request merged**